build-info-extractor/src/main/java/org/jfrog/build/extractor/packageManager/PackageManagerExtractor.java

```diff
@@ -56,6 +60,28 @@ static void saveBuildInfoToFile(ArtifactoryClientConfiguration clientConfigurati
         }
     }
 
+    /**
+     * Collect environment variables according to the env include-exclude patterns
+     *
+     * @param clientConfiguration - Artifactory client configuration
+     * @param build               - The target build
+     */
+    private static void collectEnv(ArtifactoryClientConfiguration clientConfiguration, Build build) {
```

Comment:
I think it is worth moving this method into a util, and have the maven and gradle flow use it as well, instead of the separate logic used by them. I understand that this will require some refactoring for maven and gradle, but I think the code will benefit from this. We can then test the collect env vars functionality from the Jenkins code base, to make sure we're good.